src/Symfony/Bundle/ArgumentResolver/PayloadArgumentResolver.php

```diff
@@ -18,63 +18,120 @@
 use ApiPlatform\Util\OperationRequestInitiatorTrait;
 use ApiPlatform\Util\RequestAttributesExtractor;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
 use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
 use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
 
-final class PayloadArgumentResolver implements ValueResolverInterface
-{
-    use OperationRequestInitiatorTrait;
-
-    public function __construct(
-        ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory,
-        private readonly SerializerContextBuilderInterface $serializationContextBuilder
-    ) {
-        $this->resourceMetadataCollectionFactory = $resourceMetadataCollectionFactory;
-    }
-
-    public function supports(Request $request, ArgumentMetadata $argument): bool
+if (interface_exists(ValueResolverInterface::class)) {
```

Comment:
I'm not sure we should declare 2 classes in the same file

Reply:
Yeah I wasn't sure either, I've seen both this and a more split approach before so I just went with this first. If you want I can try and separate it a bit

Reply:
if it's the same code for both (only change of interface), a new interface could be created like:

```php
if (interface_exists(ValueResolverInterface::class)) {
    /** @internal */
    interface CompatibleValueResolverInterface extends ValueResolverInterface
    {
    }
} else {
    /** @internal */
    interface CompatibleValueResolverInterface extends ArgumentValueResolverInterface
    {
    }
}
```
and implement `CompatibleValueResolverInterface` instead of `ValueResolverInterface` or `ArgumentValueResolverInterface`

Reply:
It's almost the same yeah, the thing that changes can be done with a simple check for the interface as well though